### ISSUE TITLE
colors are off, for typescript

### ISSUE THREAD

**Issue by snr2718** [projekt0n/github-nvim-theme#312]
Hi, I started using your plugin very recently. I am in love with it and really appreciate all the efforts. I wanted to submit the following issue:

## The colors are off, for typescript. 

Attaching screenshots:

1. File opened in github.com ( google chrome; mac os; dark theme).

<img width="861" alt="Screenshot 2024-01-05 at 2 55 34 PM" src="https://github.com/projekt0n/github-nvim-theme/assets/53249506/ffc2cfb1-9e40-4dd0-b7a1-30b5ba96be6b">

2. File openend in neovim ( neovim version: 0.9.4; terminal: iterm2; theme: github_dark_default)

<img width="776" alt="Screenshot 2024-01-05 at 2 55 20 PM" src="https://github.com/projekt0n/github-nvim-theme/assets/53249506/811777f7-937e-42bd-8b26-902c01716cae">

As you can see, the colors for class name and types of class properties are different from github.com.

**Comment by tmillr** [comment]
This is true. I remember this difference when revamping the highlights, and this outcome was intentional. This is essentially how JavaScript is highlighted on GitHub.com (i.e. with class names and the `constructor` keyword receiving highlights, as opposed to TypeScript where these go unhighlighted). The additional highlights are simply being borrowed/reused from js, although a little extra work could be done in order to differentiate the 2 langs and reflect precisely the quirks of GitHub.com's interlanguage highlighting differences (but personally I would consider this: a downgrade, as less consistent, as less optimal highlighting, etc.).

TLDR: alot of langs needed to be revamped at once and it was simply less work to borrow/resuse this from the JavaScript highlighting, plus I thought it was better having these highlights included for TypeScript

**Comment by Erdemus** [comment]
Hi, I have an issue that almost all themes (builtin nvchad) highlight type names as yellow for c++ with LSP. I thought that maybe because of JavaScript reused highlighting that you mentioned. Is there a way to override yellowish to whiteish somehow globally?

**Comment by niamleeson** [comment]
Are those customizable? I've noticed that const is treated like a regular keyword.

This theme
![Screenshot 2024-05-04 at 3 08 29 AM](https://github.com/projekt0n/github-nvim-theme/assets/5668478/37e37715-c96a-4f83-9a37-ea02d7b599cd)

VSCode (I changed the color for const to be more pleasant)
![Screenshot 2024-05-04 at 3 08 58 AM](https://github.com/projekt0n/github-nvim-theme/assets/5668478/35ed31d3-b780-42de-a8b4-7bd96098ce84)

**Comment by tmillr** [comment]
> Are those customizable? I've noticed that const is treated like a regular keyword.
> 
> This theme ![Screenshot 2024-05-04 at 3 08 29 AM](https://private-user-images.githubusercontent.com/5668478/327931749-37e37715-c96a-4f83-9a37-ea02d7b599cd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMDgzMzAsIm5iZiI6MTcxODEwODAzMCwicGF0aCI6Ii81NjY4NDc4LzMyNzkzMTc0OS0zN2UzNzcxNS1jOTZhLTRmODMtOWEzNy1lYTAyZDdiNTk5Y2QucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTFUMTIxMzUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzExMWVlMzljNDU0YTA4MzJiMDFkNzdiNjZmZjk5OTQ1MzU2YWVmMGZiNjIyNTk1YTg3YzNlMTRhZTkxNTMxNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.3ZhOsArvbvWCvimIeuMr_jDoaDpQGIuY_TS-8EdndDc)
> 
> VSCode (I changed the color for `const` and `this` to be more pleasant) ![Screenshot 2024-05-04 at 3 08 58 AM](https://private-user-images.githubusercontent.com/5668478/327931826-35ed31d3-b780-42de-a8b4-7bd96098ce84.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMDgzMzAsIm5iZiI6MTcxODEwODAzMCwicGF0aCI6Ii81NjY4NDc4LzMyNzkzMTgyNi0zNWVkMzFkMy1iNzgwLTQyZGUtYThiNC03YmQ5NjA5OGNlODQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MTFUMTIxMzUwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NWQwZWYwYjY1MDE2NWEzYzg0YWViYjQ4Zjk1ODM1OWE1MjcwYmQxMmE5NjM4ZDcyZjJhNDY2YWVkYmE0OTYwMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.H5x003-75XBznvCQQfnVmsdrMNhhTHptGRS26UKReDk)

---

Maybe `this` will have to be revisited and have its default hl changed, but the hl for `const` is correct as it is a keyword and GitHub highlights it the same as any other keyword:

### Ts

```ts
const t = this.method();
```

### Js

```js
const t = this.method();
```

Also, the default `ecma` treesitter query currently [captures all keywords under the same capture](https://github.com/nvim-treesitter/nvim-treesitter/blob/9a7ad2ff7a7ea81016aca2fc89c9b2c1a5365421/queries/ecma/highlights.scm#L329-L341), so I don't think changing the hl for just `const` is currently supported. However it is technically still possible to change `const` via a custom treesitter query (if you are using treesitter). For example, you could try something like this if you'd like:

```scm
;; extends
["const"] @keyword.modifier
```

...and put that in a file at `queries/ecma/highlights.scm` under any dir in your `runtimepath` (e.g. `~/.config/nvim/queries/ecma/highlights.scm` on unix). After that you can change/configure the color of this hl group via config:

```lua
require('github-theme').setup({ groups = { ['@keyword.modifier'] = { ... } } })
```

> [!NOTE]
> Replace `['@keyword.modifier']` with `['@keyword.modifier.ecma']` if you only want to change the color of modifier keywords for js and ts instead of all langs.

To clear the hl for `this` you should be able to do:

```lua
['@variable.builtin.ecma'] = {}, -- empty table
```
